### DISCUSSION TITLE
Discord buttons not working as expected

### DISCUSSION THREAD

**Discussion by NISH-Original** [Rapptz/discord.py#6964] [Q&A]
umm hello guys, I wanna make some buttons by the help of `discord_components` for a rock, paper, scissor game...
soo I have made the following code: 
```py
await ctx.send(
        "Choices:",
        components=[
            Button(style=ButtonStyle.blue, label="✊ Rock"),
            Button(style=ButtonStyle.red, label="🖐️ Paper"),
            Button(style=ButtonStyle.green, label="✌️ Scissors"),
        ],
    )
    global user_choice
    res = await ddb.wait_for_interact("button_click")
    if res.channel == ctx.message.channel:
        await res.respond(
            type=InteractionType.ChannelMessageWithSource,
            content=f'{res.component.label} clicked'
        )
    user_choice = str(res.component.label)
    await ctx.send(user_choice)
```
Soo, this creates three buttons, rock, paper, scissors, and the value of that is stored in the `user_choice` variable, and it also sends the value of the variable successfully as in the last line of the code...
But, the code does not proceed further to display the bot's choice and the algorithm to show the winner, which is shown below:
```py
global bot_choice
    bot_choice = random.choice(["✊ Rock","🖐️ Paper","✌️ Scissors"])
    winner = ""
    if user_choice == bot_choice:
        embedVar = discord.Embed(title="TIE!", description = "gg", color = 0xff9900)
        await ctx.send(embed = embedVar)
    elif user_choice == "rock":
        if bot_choice == "paper":
            winner = "bot"
        elif bot_choice == "scissors":
            winner = "user"
    elif user_choice == "paper":
        if bot_choice == "scissors":
            winner = "bot"
        elif bot_choice == "rock":
            winner = "user"
    elif user_choice == "scissors":
        if bot_choice == "rock":
            winner = "bot"
        elif bot_choice == "paper":
            winner = "user"
```
but the problem is that this part of the code is not  being run... thanks in advance!
![image](https://user-images.githubusercontent.com/75113251/119797342-2f97c680-beeb-11eb-9dad-07b491e0f245.png)

**Comment by gbdlin** [comment]
You are sending the response only on a tie, there is no message when user wins or loses in your example.

**Reply by NISH-Original** [reply]
hey! thanks for replying!! sorry I did not add it here, but I have written code which prints all the results, win, lose, or tie... 😅
Also, ig the prob was with my discord embed that I made later in the code, so when i removed it, it worked fine! 
![image](https://user-images.githubusercontent.com/75113251/119860206-27f61300-bf27-11eb-9389-5804e7078e96.png)

**Comment by Rapptz** [comment]
This is a place for help with `discord.py`, not 3rd party libraries.
